### ISSUE TITLE
custom "editor.wordWrap" value not preserved when working with .md files

### ISSUE THREAD

**Issue by jwhit-MSFT** [Microsoft/vscode#39718]
- VSCode Version: Code 1.18.1 (929bacba01ef658b873545e26034d1a8067445e9, 2017-11-16T18:32:36.026Z)
- OS Version: Windows_NT ia32 10.0.16299
- Extensions: 

Steps to Reproduce:

1. Add the following to the user settings file:
 "editor.wordWrap": "wordWrapColumn",
    "editor.wordWrapColumn": 40,

2. Note that .md files do not wrap at 40 columns. I assume this is because the special .md default is taking precedence?

Reproduces without extensions: Yes/No

**Comment by Peteraya** [comment]
Could anyone specifie where we can find the place where the "wrapping" actually happens? I'm trying to find in the code where the wrapping is being applied, but I'm not finding it.

**Comment by alexandrudima** [comment]
---

If you search for `wordWrap` in settings, by default, VS Code ships with:
```
"editor.wordWrap": "off"
...
"[markdown]":  {
    "editor.wordWrap": "on",
    ...
}
```
which means, **out of the box, word wrapping is enabled for markdown**.

---

**When opening a file that is dominated by long lines, VSCode will also word wrap**. e.g. try opening `jquery-3.2.1.min`. In such cases we show a button at the top of the editor that can be used to unwrap:

![image](https://user-images.githubusercontent.com/5047891/33121223-706033d8-cf74-11e7-8455-c9d063b4a723.png)

---

**Comment by jwhit-MSFT** [comment]
Actually, that wasn't the original bug. I want to specify a specific column size for .md files using the following in the user settings:

"editor.wordWrap": "wordWrapColumn",
"editor.wordWrapColumn": 40,

Can you confirm that this is working? It doesn't for me--it overlooks that setting with .md files in the mots recent version of VSCode.

**Comment by alexandrudima** [comment]
* https://code.visualstudio.com/docs/getstarted/settings#_language-specific-editor-settings
* https://zerokspot.com/weblog/2017/04/11/vscode-language-specific-settings/

user language-specific > default language-specific > user non-language-specific > default non-language-specific

i.e. because we ship with the default:
```
  "[markdown]":  {
    "editor.wordWrap": "on",
    "editor.quickSuggestions": false
  }
```

That default will always win over a user non-language specific setting, unless you overwrite specifically that language-specific setting. 

TL;DR: use the following in your settings:
```
  "[markdown]": {
    "editor.wordWrap": "wordWrapColumn",
    "editor.wordWrapColumn": 40
  }
```

**Comment by sandy081** [comment]
@alexandrudima I see. Adding some UX in the settings editor like blurring or squiggles to indicate that a setting is overridden is helpful?
